src/scripts/main.js

```diff
@@ -18,10 +18,34 @@ const food = {
   },
 };
 
+const treeDiv = document.createElement('div');
+
+treeDiv.id = 'tree';
+document.body.appendChild(treeDiv);
+
 const tree = document.querySelector('#tree');
 
 function createTree(element, data) {
-  // WRITE YOUR CODE HERE
+  if (element === null || data === undefined) {
+    throw new Error('arguments must not be null or undefined');
+  }
+
+  const ul = document.createElement('ul');
+
+  for (const key in data) {
+    const li = document.createElement('li');
+
+    li.textContent = key;
+
+    if (Object.entries(data[key]).length > 0) {
+      createTree(li, data[key]);
+    }
+    ul.appendChild(li);
+  }
```

Comment:
You are using a for-in loop to iterate over the properties of an object. While this is not incorrect, it can lead to issues if the object's prototype has enumerable properties. To avoid this, it's recommended to use Object.keys() or Object.entries() with a for-of loop instead.